server/db/actions/Dog.js

```diff
@@ -0,0 +1,57 @@
+import dbConnect from "../dbConnect";
+import Dog from "../models/Dog";
+import Log from "../models/Log";
+import User from "../models/User";
+
+/**
+ * Creates a new dog in the database
+ * @param {*} dogData Object that has been parsed by Zod for validity
+ * @returns string id of the dog if successfully saved, error otherwise
+ */
+export async function createDog(dogData) {
+  await dbConnect();
+
+  // recentLogs array
```

Comment:
Not sure we need to check this since logs are only created for a dog _after_ it has been created. Just make sure to remove `recentLogs` from the Zod schema.